uvm/impl/com/untangle/uvm/LocalDirectoryImpl.java

```diff
@@ -617,11 +648,23 @@ private void updateXauthSecrets(LinkedList<LocalDirectoryUser> list)
 
             auth.write(FILE_DISCLAIMER);
 
-            for (LocalDirectoryUser user : list) {
-                byte[] rawPassword = Base64.decodeBase64(user.getPasswordBase64Hash().getBytes());
-                String userPassword = new String(rawPassword);
-                auth.write(user.getUsername() + " : XAUTH 0x" + stringHexify(userPassword) + "\n");
-                auth.write(user.getUsername() + " : EAP 0x" + stringHexify(userPassword) + "\n");
+            try {
+                for (LocalDirectoryUser user : list) {
+                    String userPassword;
+                    if(!StringUtils.isBlank(user.getPasswordBase64Hash())){
+                        byte[] rawPassword = Base64.decodeBase64(user.getPasswordBase64Hash().getBytes());
+                        userPassword = new String(rawPassword);
+                    }else{
+                        userPassword = PasswordUtil.getDecryptPassword(user.getEncryptedPassword());
+                        if (userPassword == null) {
+                            throw new NullPointerException("Got null as decryptedPassword");
```

Comment:
This would skip the rest of the for loop, shouldn't we just log and continue with rest of the entries in loop, or skip entire loop , I don't know!

Reply:
Checked for the multiple user seems we should add the entries for user having correct password.